app/questionnaire/schema_utils.py

```diff
@@ -39,10 +40,22 @@ def _choose_variant(
 ):
     if block.get(single_key):
         return block[single_key]
-
     for variant in block.get(variants_key, []):
         when_rules = variant.get("when", [])
-        if evaluate_when_rules(
+
+        if isinstance(when_rules, dict):
+            when_rule_evaluator = RuleEvaluator(
+                schema,
+                answer_store,
+                list_store,
+                metadata,
+                {},
```

Comment:
Since variants can now used the new rules, we should pass in response metadata. It needs to be passed into `transform_variants` and down to this method. Where `transform_variants` is called, `response_metadata` data is available.